### PR TITLE
store/tikv: refactor copIterator

### PR DESCRIPTION
**Pull request by coocood**
Split copIterator into worker and sender, improve readablity.

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao @zhexuany PTAL

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -382,6 +382,25 @@ type copIterator struct {
...
+	finished <-chan struct{}
+}
+
+// copIteratorTaskSender sends tasks to taskCh then wait for the workers to exit.
+type copIteratorTaskSender struct {
+	taskCh   chan<- *copTask
+	wg       *sync.WaitGroup
+	tasks    []*copTask
+	finished <-chan struct{}
+	respChan chan<- copResponse
```
[the remark this answers is not in the document]
Comment (coocood) [reply store/tikv/coprocessor.go]: It is closed by sender.

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -704,7 +735,7 @@ func buildCopTasksFromRemain(bo *Backoffer, cache *RegionCache, lastRange *copro
 // split:      [s1   -->   s2)
 // In normal scan order, all data before s1 is consumed, so the remain ranges should be [s1 --> r2) [r3 --> r4)
 // In reverse scan order, all data after s2 is consumed, so the remain ranges should be [r1 --> r2) [r3 --> s2)
-func calculateRemain(ranges *copRanges, split *coprocessor.KeyRange, desc bool) *copRanges {
+func (worker *copIteratorWorker) calculateRemain(ranges *copRanges, split *coprocessor.KeyRange, desc bool) *copRanges {
```
[the remark this answers is not in the document]
Comment (coocood) [reply store/tikv/coprocessor.go]: It's more readable as a method, so we know how it is used.
And we can set it to function if it's used somewhere else.

File: store/tikv/coprocessor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply store/tikv/coprocessor.go]: It close `taskCh`, `respChan` is not used in `copIteratorTaskSender`
https://github.com/pingcap/tidb/pull/6225/files#diff-c27388ffb48c6f6eaeefe07dd3243530R474

File: store/tikv/coprocessor.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply store/tikv/coprocessor.go]: ok

File: store/tikv/coprocessor.go [hunk 3]
```diff
@@ -382,6 +382,25 @@ type copIterator struct {
 	wg       sync.WaitGroup
 }
 
+// copIteratorWorker receive tasks from copIteratorTaskSender and handle tasks, sent the copResponse to respChan.
```
[the remark this answers is not in the document]
Comment (zhexuany) [reply store/tikv/coprocessor.go]: @coocood any update?

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**